From f9e8ed5472c91cd2ee5d4ffb50c9c611ce59e0cb Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Tue, 26 Nov 2024 19:25:17 +0530
Subject: [PATCH] firmware: scm: Adding support for tzapp log version 2

Adding support for tzapp log version 2 for IPQ54xx target

Change-Id: I368e9e4420161c163fcb977653201b275a69625a
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 include/linux/firmware/qcom/qcom_scm.h | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/include/linux/firmware/qcom/qcom_scm.h b/include/linux/firmware/qcom/qcom_scm.h
index 6bd40c0f01ae..509ae71e3b62 100644
--- a/include/linux/firmware/qcom/qcom_scm.h
+++ b/include/linux/firmware/qcom/qcom_scm.h
@@ -152,16 +152,26 @@ struct qsee_reg_log_buf_req {
 	uint64_t len;
 };
 
-struct tzdbg_log_pos_t {
+struct tzdbg_log_v1_pos_t {
 	uint16_t wrap;
 	uint16_t offset;
 };
 
-struct qtidbg_log_t {
-	struct tzdbg_log_pos_t log_pos;
+struct qtidbg_log_v1_t {
+	struct tzdbg_log_v1_pos_t log_pos;
 	uint8_t log_buf[];
 };
 
+struct tzdbg_log_v2_pos_t {
+	u32 wrap;
+	u32 offset;
+};
+
+struct qtidbg_log_v2_t {
+	struct tzdbg_log_v2_pos_t log_pos;
+	u8 log_buf[];
+};
+
 struct qcom_scm_hdcp_req {
 	u32 addr;
 	u32 val;
-- 
2.34.1

